core/src/main/java/org/fao/geonet/util/XslUtil.java

```diff
@@ -989,6 +995,60 @@ public static String getLanguage() {
         }
     }
 
+    /**
+     * Build data URL like data:image/png;base64, iVBORw...
+     */
+    public static String buildDataUrl(String url, Integer size) {
+        StringBuilder sb = new StringBuilder("data:");
+        String supportedExtension = "jpg|jpeg|png|gif|tif|tiff";
+
+        String extension = Files.getFileExtension(url);
+        if (extension.matches(supportedExtension)) {
+
+            try {
+                Matcher m = Pattern.compile(".*/api/records/(.*)/attachments/(.*)$").matcher(url);
+                BufferedImage image;
+                if (m.find()) {
+                    Store store = ApplicationContextHolder.get().getBean(FilesystemStore.class);
+                    try (Store.ResourceHolder file = store.getResourceInternal(
+                        m.group(1),
+                        MetadataResourceVisibility.PUBLIC,
+                        m.group(2), true)) {
+                        image = ImageIO.read(file.getPath().toFile());
+                    }
+                } else {
+                    image = ImageIO.read(new URL(url));
+                }
+
+                if (image != null) {
```

Comment:
If using CMIS or S3, I think `image` will be `null` here, maybe to check if null to use the full url.

Reply:
Solved in https://github.com/geonetwork/core-geonetwork/pull/5467/commits/c7062c2a37f7a68b7b21040e0255c6d9003166c8 with an alternative solution.